File: app/src/main/java/com/edurda77/filmlibrary/tmp/SectionListDataAdapter.java
package com.edurda77.filmlibrary.tmp;

import android.content.Context;

import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import com.edurda77.filmlibrary.R;
import com.edurda77.filmlibrary.ui.FilmActivity;
import com.edurda77.filmlibrary.domain.Movie;

import java.util.ArrayList;


public class SectionListDataAdapter extends RecyclerView.Adapter<SectionListDataAdapter.SingleItemRowHolder> {
    interface OnStateClickListener{
        void onStateClick(Movie movie, int position);
    }
    private ArrayList<Movie> itemsList;
    private Context mContext;
    private final OnStateClickListener onClickListener;

    public SectionListDataAdapter(Context context, ArrayList<Movie> itemsList, OnStateClickListener onClickListener) {
        this.onClickListener = onClickListener;
        this.itemsList = itemsList;
        this.mContext = context;
    }

    @Override
    public SingleItemRowHolder onCreateViewHolder(ViewGroup viewGroup, int i) {
        View v = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.plaiyng_film, null);
        SingleItemRowHolder mh = new SingleItemRowHolder(v);
        return mh;
    }

    @Override
    public void onBindViewHolder(SingleItemRowHolder holder, int i) {

        Movie movie = itemsList.get(i);

        holder.titleMovie.setText(movie.getMovieTitle());
        holder.yearMovie.setText(movie.getYear());
        holder.rangMovie.setText((int) movie.getRang());
        holder.itemView.setOnClickListener(v -> onClickListener.onStateClick(movie, i));

       /* Glide.with(mContext)
                .load(feedItem.getImageURL())
                .diskCacheStrategy(DiskCacheStrategy.ALL)
                .centerCrop()
                .error(R.drawable.bg)
                .into(feedListRowHolder.thumbView);*/
    }

    @Override
    public int getItemCount() {
        return (null != itemsList ? itemsList.size() : 0);
    }

    public class SingleItemRowHolder extends RecyclerView.ViewHolder {

        protected TextView titleMovie;
        protected TextView yearMovie;
        protected TextView rangMovie;
        //protected ImageView itemImage;


        public SingleItemRowHolder(View view) {
            super(view);

            this.titleMovie = view.findViewById(R.id.title_movie);
            this.yearMovie = view.findViewById(R.id.year_movie);
            this.rangMovie = view.findViewById(R.id.rang_movie);
            //this.itemImage = (ImageView) view.findViewById(R.id.itemImage);




        }

    }

}

File: app/src/main/java/com/edurda77/filmlibrary/tmp/SectionListDataAdapter.kt
package com.edurda77.filmlibrary.tmp

import android.content.Context

import androidx.recyclerview.widget.RecyclerView
import com.edurda77.filmlibrary.tmp.SectionListDataAdapter.SingleItemRowHolder
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import com.edurda77.filmlibrary.R
import android.widget.TextView
import com.edurda77.filmlibrary.domain.Movie
import java.util.ArrayList

class SectionListDataAdapter(
    private val mContext: Context,
    private val itemsList: ArrayList<Movie>?,
    private val onClickListener: OnStateClickListener
) : RecyclerView.Adapter<SingleItemRowHolder>() {
    interface OnStateClickListener {
        fun onStateClick(movie: Movie?, position: Int)
    }

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): SingleItemRowHolder {
        val v =
            LayoutInflater.from(viewGroup.context).inflate(R.layout.plaiyng_film, null)
        return SingleItemRowHolder(v)
    }

    override fun onBindViewHolder(holder: SingleItemRowHolder, i: Int) {
        val movie = itemsList!![i]
        holder.titleMovie.text = movie.movieTitle
        holder.yearMovie.setText(movie.year)
        holder.rangMovie.setText(movie.rang.toInt())
        holder.itemView.setOnClickListener { v: View? -> onClickListener.onStateClick(movie, i) }

        /* Glide.with(mContext)
                .load(feedItem.getImageURL())
                .diskCacheStrategy(DiskCacheStrategy.ALL)
                .centerCrop()
                .error(R.drawable.bg)
                .into(feedListRowHolder.thumbView);*/
    }

    override fun getItemCount(): Int {
        return itemsList?.size ?: 0
    }

    inner class SingleItemRowHolder(view: View) : RecyclerView.ViewHolder(view) {
        var titleMovie: TextView
        var yearMovie: TextView
        var rangMovie: TextView

        //protected ImageView itemImage;
        init {
            titleMovie = view.findViewById(R.id.title_movie)
            yearMovie = view.findViewById(R.id.year_movie)
            rangMovie = view.findViewById(R.id.rang_movie)
            //this.itemImage = (ImageView) view.findViewById(R.id.itemImage);
        }

    }


}